drivers/net/wireless/mediatek/mt76/mt7921/debugfs.c

```diff
@@ -159,6 +159,80 @@ mt7921_queues_read(struct seq_file *s, void *data)
 	return 0;
 }
 
+static void mt7921_seq_puts_array(struct seq_file *file, const char *str,
+			 s8 *val, int len)
+{
+	int i;
+
+	seq_printf(file, "%-16s:", str);
+	for (i = 0; i < len; i++)
+		if (val[i] == 127)
+			seq_printf(file, " %6s", "N.A");
+		else
+			seq_printf(file, " %6d", val[i]);
+	seq_puts(file, "\n");
+}
+
+
+#define mt7921_print_txpwr_entry(prefix, rate)					\
+({									\
+	mt7921_seq_puts_array(s, #prefix " (user)", txpwr.data[TXPWR_USER].rate, \
+			    ARRAY_SIZE(txpwr.data[TXPWR_USER].rate)); \
+	mt7921_seq_puts_array(s, #prefix " (eeprom)", txpwr.data[TXPWR_EEPROM].rate, \
+			    ARRAY_SIZE(txpwr.data[TXPWR_EEPROM].rate));	\
+	mt7921_seq_puts_array(s, #prefix " (tmac)", txpwr.data[TXPWR_MAC].rate,	\
+			    ARRAY_SIZE(txpwr.data[TXPWR_MAC].rate));	\
+})
+
+static int
+mt7921_txpwr(struct seq_file *s, void *data)
+{
+	struct mt7921_dev *dev = dev_get_drvdata(s->private);
+	struct mt7921_txpwr txpwr;
+	int ret;
+
+	ret = mt7921_get_txpwr_info(dev, &txpwr);
```

Comment:
I think you should check the hw is not in doze mode here

Reply:
ack